qradar_connector.py

```diff
@@ -890,33 +879,51 @@ def _report_back(self, new_offenses, runs):
             error_msg = self._get_error_message_from_exception(e)
             self.debug_print('Error occurred: {}'.format(error_msg))
 
-    def _create_offense_artifacts(self, offense, container_id, offense_id, is_events_artifact=False, is_events=True):
+    def _create_offense_artifacts(self, offense, container_id, offense_id):
         """ This function is used to create artifacts in given container using finding data.
 
         :param offense: Data of single finding
         :param container_id: ID of container in which we have to create the artifacts
         :param offense_id: ID of the offense
-        :param is_events_artifact: Boolean flag to verify the calling method
-        :param is_events: Boolean flag to check the length of events
         :return: status(success/failure), message
         """
 
-        artifact = {}
-        artifact['name'] = 'Offense Artifact'
-        artifact['container_id'] = container_id
-        artifact['source_data_identifier'] = offense_id
-        artifact['cef'] = offense
-        artifact['label'] = 'offense'
-        create_artifact_status = None
-        create_artifact_msg = None
-        if is_events_artifact:
-            if not is_events:
-                artifact['run_automation'] = True
-            create_artifact_status, create_artifact_msg, _ = self.save_artifact(artifact)
-        else:
-            create_artifact_status, create_artifact_msg, _ = self.save_artifacts([artifact])
-        if phantom.is_fail(create_artifact_status):
-            return phantom.APP_ERROR, create_artifact_msg
+        artifact = {
+            'name': 'Offense Artifact',
+            'container_id': container_id,
+            'source_data_identifier': offense_id,
+            'cef': offense,
+            'label': 'offense',
+            'run_automation': True
+        }
+
+        # This method will be called in the loop, add offense artifact only if it is not created earlier.
+        # At least one artifact in the list indicates that the offense artifact is already created.
+        if not self._artifacts_list:
+            self._artifacts_list.append(artifact)
+
+    def _ingest_collected_artifacts(self, offense_id):
```

Comment:
Seeing that this method contains some complicated logic, would it be possible to add some docstring to help us understand this block faster in the future?

Reply:
Added a docstring and comment for artifact creation and ingest function.